src/components/Tooltip/TooltipRenderedOnPageBody.js

```diff
@@ -118,36 +129,51 @@ class TooltipRenderedOnPageBody extends React.PureComponent {
             this.props.maxWidth,
             this.state.tooltipWidth,
             this.state.tooltipHeight,
-            this.state.tooltipTextWidth,
+            this.state.tooltipContentWidth,
             this.props.shiftHorizontal,
             this.props.shiftVertical,
         );
-        return ReactDOM.createPortal(
-            <Animated.View
-                onLayout={this.measureTooltip}
-                style={[tooltipWrapperStyle, animationStyle]}
-            >
+
+        let content;
+        const contentRef = (ref) => {
+            // Once the content for the tooltip first renders, update the width of the tooltip dynamically to fit the width of the content.
+            // Note that we can't have this code in componentDidMount because the ref for the content won't be set until after the first render
+            if (this.contentRef) {
+                return;
+            }
+
+            this.contentRef = ref;
+            this.updateTooltipContentWidth();
+        };
+
+        if (this.props.renderTooltipContent) {
+            content = (
+                <View ref={contentRef}>
+                    {this.props.renderTooltipContent()}
+                </View>
+            );
+        } else {
+            content = (
                 <Text numberOfLines={this.props.numberOfLines} style={tooltipTextStyle}>
-                    <Text
-                        style={tooltipTextStyle}
-                        ref={(ref) => {
-                            // Once the text for the tooltip first renders, update the width of the tooltip dynamically to fit the width of the text.
-                            // Note that we can't have this code in componentDidMount because the ref for the text won't be set until after the first render
-                            if (this.textRef) {
-                                return;
-                            }
-
-                            this.textRef = ref;
-                            this.updateTooltipTextWidth();
-                        }}
-                    >
+                    <Text style={tooltipTextStyle} ref={contentRef}>
                         {this.props.text}
                     </Text>
                 </Text>
-                <View style={pointerWrapperStyle}>
-                    <View style={pointerStyle} />
-                </View>
-            </Animated.View>,
+            );
+        }
+
+        return ReactDOM.createPortal(
+            <>
+                <Animated.View
+                    onLayout={this.measureTooltip}
+                    style={[tooltipWrapperStyle, animationStyle]}
+                >
+                    {content}
+                    <View style={pointerWrapperStyle}>
+                        <View style={pointerStyle} />
+                    </View>
+                </Animated.View>
+            </>,
```

Comment:
```suggestion
            <Animated.View
                onLayout={this.measureTooltip}
                style={[tooltipWrapperStyle, animationStyle]}
            >
                {content}
                <View style={pointerWrapperStyle}>
                    <View style={pointerStyle} />
                </View>
            </Animated.View>,
```

Remove unnecessary fragment.